### PR TITLE
Remove Simulation() and pulser.simulation

### PR DESCRIPTION
**Pull request by HGSilveri**
- Removes the deprecated `Simulation()` class
- Removes the legacy `pulser.simulation` module

### CODE REVIEW

File: pulser-core/pulser/sequence/_seq_drawer.py [hunk 1]
```diff
@@ -698,7 +698,7 @@ def phase_str(phi: Any) -> str:
                         alpha=0.3,
                         hatch="////",
                     )
-                else:
+                else:  # pragma: no cover
```
Comment (a-corni) [inline pulser-core/pulser/sequence/_seq_drawer.py:701]: Why is this needed ?

**Review by a-corni — COMMENTED** [review]
Just one nit :)

File: pulser-core/pulser/sequence/_seq_drawer.py (hunk 1 above)
Comment (HGSilveri) [reply pulser-core/pulser/sequence/_seq_drawer.py:701]: Basically this case is only touched when using `draw_sequence()` with modulation and a sampling rate. While `Simulation()` was doing this, `QutipEmulator()` uses `draw_samples()`. I could delete it but it might come in handy again in the future, so I just kept it

Comment (a-corni) [reply pulser-core/pulser/sequence/_seq_drawer.py:701]: Sure, let's keep it then ;)

**Review by a-corni — APPROVED** [review]
